ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -651,6 +1107,51 @@ public void deleteDataset(String accountName, String containerName, String datas
     }
   }
 
+  void updateAccountsWithMySqlStoreNew(Collection<Account> accounts) throws SQLException {
+    long startTimeMs = System.currentTimeMillis();
+    logger.trace("Start updating accounts={} into new MySql DB", accounts);
+
+    // Get account and container changes info
+    List<AccountUpdateInfo> accountsUpdateInfo = new ArrayList<>();
+
+    for (Account account : accounts) {
+      boolean isAccountAdded = false, isAccountUpdated = false;
+      List<Container> addedContainers;
+      List<Container> updatedContainers = new ArrayList<>();
+      Account accountInCacheOld = getAccountByIdOld(account.getId());
+      Account accountInCacheNew = getAccountByIdNew(account.getId());
+      //only when the account in old cache but not exist in new cache, we need to upload account from old cache to new db
+      //and refresh new cache. After migration is done, we can remove this line.
+      if (accountInCacheNew == null && accountInCacheOld != null) {
+        accountServiceMetrics.updateAccountFromOldCacheToNewDbCount.inc();
+        uploadAccountFromOldCacheToNewDBAndRefreshNewCache(accountInCacheOld);
+      }
+      //refresh accountInCacheNew.
+      accountInCacheNew = getAccountByIdNew(account.getId());
+      //after above action, there's no chance the account exist in old db only.
+      if (accountInCacheNew == null) {
```

Comment:
As the comment says, accountInCacheNew should never be null. Will this condition be ever true?

Reply:
If user provide new account, this accountInCacheNew should be null. the comment just explain if for the old db the accountInCache != null, then accountInCacheNew should not be null.